src/main/java/build/buildfarm/common/CommandUtils.java

```diff
@@ -34,4 +37,50 @@ public static boolean isTest(Command command) {
     return command.getEnvironmentVariablesList().stream()
         .anyMatch((envVar) -> envVar.getName().equals("XML_OUTPUT_FILE"));
   }
+
+  /**
+   * @brief Get the command's output paths.
+   * @details This abstracts over REAPI changes around how this information is encoded in a Command.
+   * @param command The command to get output paths from.
+   * @param actionRoot The action root used to resolve the Command's relative paths.
+   * @return The list of output paths.
+   * @note Suggested return identifier: output_paths.
+   */
+  public static List<Path> getResolvedOutputPaths(Command command, Path actionRoot) {
+
+    // REAPI clients previously needed to specify whether the output path was a directory or file.
+    // This turned out to be too restrictive-- some build tools don't know what an action produces
+    // until it is done.
+    // https://github.com/bazelbuild/remote-apis/issues/75
+    // That's unfortunate for server implementations, because now buildfarm has to distinguish
+    // between files and paths itself.
+
+    // To quote REAPI:
+    // "New in v2.1: this field supersedes the DEPRECATED `output_files` and
+    // `output_directories` fields. If `output_paths` is used, `output_files` and
+    // `output_directories` will be ignored!"
+    // https://github.com/bazelbuild/remote-apis/blob/3a21deee813d0b98aaeef9737c720e509e10dc8b/build/bazel/remote/execution/v2/remote_execution.proto#L651-L653
+    List<Path> resolvedPaths = new ArrayList<>();
```

Comment:
Small nit here: if getOutputPathsCount() != 0, just return the getOutputPathsList().

Reply:
fixed